include/sound/soc-acpi.h

```diff
@@ -193,6 +194,14 @@ struct snd_soc_acpi_link_adr {
  *  is not constant since this field may be updated at run-time
  * @sof_tplg_filename: Sound Open Firmware topology file name, if enabled
  * @tplg_quirk_mask: quirks to select different topology files dynamically
+ * @get_tplg_files: to get sub-topology file names. The return value is the number
```

Comment:
this is optional callback, if specified then instead of the single `sof_tplg_filename` the callback will return the list of function topology files to be loaded.

I would also rename the `get_tplg_files` to `get_function_topology_files` to make things a bit cleaner ( I could not think a better name for soc-acpi-intel-get-tplg.c , perhaps sof-function-topology[-lib].c ?

Reply:
Consider to the callback name length, I will rename `get_tplg_files` to `get_sub_tplg_files`.